src/Microsoft.Azure.SignalR/EndpointProvider/ServiceEndpointManager.cs

```diff
@@ -30,5 +44,74 @@ public override IServiceEndpointProvider GetEndpointProvider(ServiceEndpoint end
 
             return new ServiceEndpointProvider(endpoint, _options);
         }
+
+        private void OnChange(ServiceOptions options)
+        {
+            Log.DetectConfigurationChanges(_logger);
+
+            // Reset local cache and validate result
+            SetValuableEndpoints(GetEndpoints(options));
+            if (Endpoints.Length == 0)
+            {
+                throw new ArgumentException(ServiceEndpointProvider.ConnectionStringNotFound);
+            }
+            var updatedEndpoints = Endpoints.ToList();
+
+            var endpoints = GetChangedEndpoints(updatedEndpoints);
+
+            // RenameServiceEndpoints(endpoints.RenamedEndpoints);
+
+            _ = DoScaleAsync(endpoints.AddedEndpoints, endpoints.RemovedEndpoints);
+
+            _endpointsStore = updatedEndpoints.ToList();
+        }
+
+        private Task DoScaleAsync(IReadOnlyList<ServiceEndpoint> addedEndpoints, IReadOnlyList<ServiceEndpoint> removedEndpoints)
+        {
+            // First add then remove to minor the affect to new clients.
+            // Update EndpointType follow same process.
+            // AddServiceEndpoints(addedEndpoints);
+            // 
+            // RemoveServiceEndpoint(removedEndpoints);
```

Comment:
>RemoveServiceEndpoint [](start = 15, length = 21)

if it is not enabled, shall we move it and its implementation to another PR with tests ready?